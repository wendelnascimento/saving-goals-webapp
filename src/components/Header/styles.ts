import styled from 'styled-components';

export const Bar = styled.header`
  max-width: 100vw;
  width: 100%;
  background: ${({ theme }) => theme.white};
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid ${({ theme }) => theme.superlightGray};
`;
