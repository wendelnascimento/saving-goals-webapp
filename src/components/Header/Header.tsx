import React from 'react';

import Logo from '../Logo';

import { Bar, BackButton, ButtonImage } from './styles';

import arrowSvg from '../../icons/arrow.svg';

const Header = () => (
  <Bar>
    <BackButton>
      <ButtonImage src={arrowSvg} />
    </BackButton>
    <Logo />
  </Bar>
);

export default Header;
