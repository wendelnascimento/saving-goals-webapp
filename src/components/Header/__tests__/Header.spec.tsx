import React from 'react';
import { shallow, mount } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  it('Should render', () => {
    shallow(<Header />);
  });

  it('Should math stored snapshot', () => {
    const wrapper = mount(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
