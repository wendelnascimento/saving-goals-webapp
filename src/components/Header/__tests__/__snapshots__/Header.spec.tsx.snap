// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should math stored snapshot 1`] = `
.c4 {
  width: 100px;
}

.c0 {
  max-width: 100vw;
  width: 100%;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid;
}

.c1 {
  display: none;
}

.c2 {
  width: 6px;
}

@media screen and (max-width:550px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c0 > .c3 {
    position: absolute;
  }
}

@media screen and (max-width:550px) {
  .c1 {
    display: block;
    margin-right: auto;
    display: block;
    width: 40px;
    height: 40px;
    padding: 8px;
  }
}

<Header>
  <styled.header>
    <header
      className="c0"
    >
      <styled.button>
        <button
          className="c1"
        >
          <styled.img
            src="test-file-stub"
          >
            <img
              className="c2"
              src="test-file-stub"
            />
          </styled.img>
        </button>
      </styled.button>
      <Logo>
        <styled.img
          src="test-file-stub"
        >
          <img
            className="c3 c4"
            src="test-file-stub"
          />
        </styled.img>
      </Logo>
    </header>
  </styled.header>
</Header>
`;
